generators/server/generator.mjs

```diff
@@ -223,19 +205,26 @@ spring:
         );
       },
 
-      async mainClass({ application: { baseName, packageFolder, databaseTypeSql, reactive } }) {
+      async mainClass({ application: { baseName, packageFolder, databaseTypeSql, prodDatabaseTypePostgres, reactive } }) {
         const mainClassPath = `${SERVER_MAIN_SRC_DIR}${packageFolder}/${this.getMainClassName(baseName)}.java`;
         let content = this.readDestination(mainClassPath);
-        const types = ['org.HdrHistogram.Histogram.class', 'org.HdrHistogram.ConcurrentHistogram.class'];
+        const types = [
+          'org.HdrHistogram.Histogram.class',
+          'org.HdrHistogram.ConcurrentHistogram.class',
+          // Required by *ToMany relationships
+          'java.util.HashSet.class',
+        ];
         const typeNames = [];
         if (databaseTypeSql) {
           types.push(
             'liquibase.configuration.LiquibaseConfiguration.class',
             'com.zaxxer.hikari.HikariDataSource.class',
             'liquibase.change.core.LoadDataColumnConfig.class',
-            'tech.jhipster.domain.util.FixedPostgreSQL10Dialect.class',
             'org.hibernate.type.TextType.class'
           );
+          if (prodDatabaseTypePostgres && !reactive) {
+            types.push('tech.jhipster.domain.util.FixedPostgreSQL10Dialect.class');
+          }
```

Comment:
I can confirm that this is only needed for Spring MVC. Seems strange, but it's true. https://github.com/mraible/spring-native-examples/pull/6/commits/86b54a822d1c3e34bb970335269ecb5d1c339430

Reply:
I think FixedPostgreSQL10Dialect is the hibernate dialect.
Hibernate is not used at reactive.